### PR TITLE
recovery timeout option for interfaces and bonds

### PR DESCRIPTION
**Pull request by vilnitsky**
support recovery-timeout parameter in interfaces and bonds
   
2 new API:
```
PUT /switches/<hostname>/interfaces/<path:interface_id>/recovery-timeout
PUT /switches/<hostname>/bonds/<bond_number>/recovery-timeout
```

### CODE REVIEW

**Comment by vilnitsky** [comment]
Hey @fbouliane, can I have your help with this one to review?

File: netman/adapters/switches/cached.py [hunk 1]
```diff
@@ -316,6 +316,14 @@ def unset_interface_lacp_force_up(self, interface_id):
         self.real_switch.unset_interface_lacp_force_up(interface_id)
         self.interfaces_cache[interface_id].force_up = None
 
+    def set_interface_recovery_timeout(self, interface_id, recovery_timeout):
+        self.real_switch.set_interface_lacp_force_up(interface_id, recovery_timeout)
```
Comment (fbouliane) [inline netman/adapters/switches/cached.py:320]: s/set_interface_lacp_force_up/set_interface_recovery_timeout/

File: netman/adapters/switches/cached.py [hunk 2]
```diff
@@ -316,6 +316,14 @@ def unset_interface_lacp_force_up(self, interface_id):
         self.real_switch.unset_interface_lacp_force_up(interface_id)
         self.interfaces_cache[interface_id].force_up = None
 
+    def set_interface_recovery_timeout(self, interface_id, recovery_timeout):
```
Comment (fbouliane) [inline netman/adapters/switches/cached.py:319]: Are you missing the unset ? Should it be cleaned during interface de-configuration ?

File: netman/adapters/switches/cached.py [hunk 3]
```diff
@@ -316,6 +316,14 @@ def unset_interface_lacp_force_up(self, interface_id):
...
+        self.real_switch.set_interface_lacp_force_up(interface_id, recovery_timeout)
+        self.interfaces_cache[interface_id].recovery_timeout = recovery_timeout
+
+    def set_bond_recovery_timeout(self, interface_id, recovery_timeout):
+        self.real_switch.set_bond_lacp_force_up(interface_id, recovery_timeout)
```
Comment (fbouliane) [inline netman/adapters/switches/cached.py:324]: s/set_bond_lacp_force_up/set_bond_recovery_timeout/

**Review by lhopps — APPROVED** [review]

File: netman/adapters/switches/cached.py (hunk 2 above)
Comment (vilnitsky) [reply netman/adapters/switches/cached.py:319]: thank you! @fbouliane , I have updated the cached interfaces

regarding unset, had this concern - so I had verified with nico it cleans up everything along with other options in reclaiming

ideally we could have a separate api to remove specifically this parameter, but not required by anything else at this point

**Pull request merged**